src/d3/d3Renderer.js

```diff
@@ -207,35 +202,24 @@ var d3Renderer = function (arg) {
       return;
     }
 
-    var layer = m_this.layer(),
-        map = layer.map(),
+    var layer = m_this.layer();
+
+    var map = layer.map(),
         upperLeft = map.gcsToDisplay(m_corners.upperLeft, null),
         lowerRight = map.gcsToDisplay(m_corners.lowerRight, null),
         center = map.gcsToDisplay(m_corners.center, null),
         group = getGroup(),
-        canvas = m_this.canvas(),
         dx, dy, scale, rotation, rx, ry;
 
-    if (canvas.attr('scale') !== null) {
-      scale = parseFloat(canvas.attr('scale') || 1);
-      rx = (parseFloat(canvas.attr('dx') || 0) +
-            parseFloat(canvas.attr('offsetx') || 0));
-      ry = (parseFloat(canvas.attr('dy') || 0) +
-            parseFloat(canvas.attr('offsety') || 0));
-      rotation = parseFloat(canvas.attr('rotation') || 0);
-      dx = scale * rx + map.size().width / 2;
-      dy = scale * ry + map.size().height / 2;
-    } else {
-      scale = Math.sqrt(
-        Math.pow(lowerRight.y - upperLeft.y, 2) +
-        Math.pow(lowerRight.x - upperLeft.x, 2)) / m_diagonal;
-      // calculate the translation
-      rotation = map.rotation();
-      rx = -m_width / 2;
-      ry = -m_height / 2;
-      dx = scale * rx + center.x;
-      dy = scale * ry + center.y;
-    }
+    scale = Math.sqrt(
```

Comment:
Before, we computed a scale for each layer of tiles and modified it upon movement.  Now, we only calculate one transform (rather than one per layer).  This is done initially and on movement (not every render).  So we actually calculate less transforms than we did.